Source/Common/schema/extensions/SVRF_animated_material/SvrfAnimatedMaterialExtension.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  SvrfAnimatedMaterialExtension.swift
+//  GLTFSceneKit_macOS
+//
+//  Created by Roman Chaley on 8/29/19.
+//
+
+import Foundation
+import SceneKit
+
+protocol SvrfTextureAnimator {
+    func animateTexture(_ model: TextureAnimationModel) throws
+}
+
+struct SvrfAnimatedMaterialExtension: GLTFCodable {
+    struct SvrfAnimatedTextureSchema: Codable {
+        let images: [Int]
+        let fps: Int?
+        
+        private enum CodingKeys: String, CodingKey {
+            case images
+            case fps
+        }
+    }
+
+    struct SvrfAnimatedPrimitiveSchema: Codable {
+        let baseColorTexture: SvrfAnimatedTextureSchema?
+        let metallicRoughnessTexture: SvrfAnimatedTextureSchema?
+        let normalTexture: SvrfAnimatedTextureSchema?
+        let occlusionTexture: SvrfAnimatedTextureSchema?
+        let emissiveTexture: SvrfAnimatedTextureSchema?
+        
+        private enum CodingKeys: String, CodingKey {
+            case baseColorTexture
+            case metallicRoughnessTexture
+            case normalTexture
+            case occlusionTexture
+            case emissiveTexture
+        }
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case data = "SVRF_animated_material"
+    }
+    
+    private enum AnimatedMaterialError: Error {
+        case noImages
+        case missingTexture
+    }
+    
+    let defaultFps = 10
```

Comment:
```suggestion
    let defaultFps = 12
```

12 seems like a "correct" lower bound fps?

Reply:
I think we had it 10 for the viewport extension